### PR TITLE
Support of UPFDeployment as configref for SMF controller

### PR DESCRIPTION
**Pull request by s3wong**

### CODE REVIEW

File: controllers/smf/resources_test.go [hunk 1]
```diff
@@ -151,16 +152,60 @@ func TestCreateDeployment(t *testing.T) {
...
+	ref := &refv1alpha1.ConfigRef{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "free5gc-upf-1",
+		},
+		Spec: refv1alpha1.ConfigRefSpec{
+			GVKC: refv1alpha1.GVKC{
+				Group:   "workload.nephio.org",
+				Version: "v1alpha1",
+				Kind:    "UPFDeployment",
+				Config:  `{"capacity":{"maxUplinkThroughput":"1G","maxDownlinkThroughput":"5G","maxSessions":1000, "maxSubscribers":1000,"maxNFConnections":2000},"interfaces":[{"name":"n6","ipv4":{"address":"10.10.12.10/24","gateway":"10.10.12.1"}},{"name":"n3","ipv4":{"address":"10.10.10.10/24","gateway":"10.10.10.1"}},{"name":"n4","ipv4":{"address":"10.10.11.10/24","gateway":"10.10.11.1"}}],"networkInstances":[{"name":"vpc-internet","interfaces":["n6"],"dataNetworks":[{"name":"apn-test","pool":[{"prefix":"100.100.0.0/16"}]}]}]}`,
```
Comment (debeaueric) [inline controllers/smf/resources_test.go:166]: This is not aligned with the global topology used where we are using subnet in 172 ranges

Comment (s3wong) [reply controllers/smf/resources_test.go:166]: @debeaueric 

That is unit test code. For e2e, the configref for SMFDeployment CR will be generated by specialization code

Comment (debeaueric) [reply controllers/smf/resources_test.go:166]: ok

**Pull request merged**